packages/workbox-window/src/Workbox.ts

```diff
@@ -195,6 +195,27 @@ class Workbox extends WorkboxEventTarget {
     return this._registration;
   }
 
+  /**
+   * Checks for updates of the registered service worker.
+   */
+  async update() {
+      if (!this._registration) {
+        if (process.env.NODE_ENV !== 'production') {
+        logger.error('Cannot update a Workbox instance without ' +
+            'being registered. Register the Workbox instance first.');
+          }
+        return;
+      }
+
+    // Reset the registration time and update count so it's not treated as 
+    // external in the `this._onUpdateFound` heuristic.
```

Comment:
I'm not sure resetting these here is what we want to do. Based on how I describe [the intention of the "external" events in the workbox-window guide](https://developers.google.com/web/tools/workbox/modules/workbox-window#when_an_unexpected_version_of_the_service_worker_is_found), an update found via calling `update()` is (for all intents and purposes) the same as an update found via the app being updated in a new tab.

Do you disagree? Based on what you said in this comment I thought you wanted to treat them the same way: https://github.com/GoogleChrome/workbox/issues/2130#issuecomment-513350467

Reply:
I changed it now but I don't understand why an update via `update()` is the same as an update via a new tab? Isn't the update eventually the same as a `register()` which is called after the page has been loaded?

Reply:
> Isn't the update eventually the same as a `register()` which is called after the page has been loaded?

Not quite. The difference is `register()` is always called first, and after the `register()` call you either have an update or you don't. But when you have an update from the `register()` call (in the same tab), it's always for a SW that is older (or the same age) as the code running on the window.

But if an update is found when calling `update()` (or `.register()` in another tab in the future), it means the updated SW is newer than the code running on the page, which means there could potentially be conflicts, depending on how old the code running on the page is.